File: MarriageCalculator/MarriageCalculator.Core/Models/MarriageGameScore.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MarriageCalculator.Core.Models;

public class MarriageGameScore
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [BsonRepresentation(BsonType.ObjectId)]
    public string MarriageGameId { get; set; } = string.Empty;

    [BsonRepresentation(BsonType.ObjectId)]
    public string PlayerId { get; set; } = string.Empty;

    public bool Seen { get; set; } = false;

    public bool Playing { get; set; } = false;

    public int Maal { get; set; } = 0;

    public int BonusPoint { get; set; } = 0;

    public bool Duply { get; set; } = false;

    public bool Winner { get; set; } = false;

    public int Score { get; set; } = 0;

    public double MoneyWon { get; set; }

    public bool Deal { get; set; } = false;

    public int Position { get; set; } = 0;

    [BsonIgnore]
    public MarriageGame? MarriageGame { get; set; }
}

File: MarriageCalculator/Android/app/src/main/java/np/com/sanjeeb/marriagecalculator/data/model/MarriageGameScore.kt
package np.com.sanjeeb.marriagecalculator.data.model

import com.google.gson.annotations.SerializedName

data class MarriageGameScore(
    @SerializedName("id") val id: String = "",
    @SerializedName("marriageGameId") val marriageGameId: String = "",
    @SerializedName("playerId") val playerId: String = "",
    @SerializedName("seen") val seen: Boolean = false,
    @SerializedName("playing") val playing: Boolean = false,
    @SerializedName("maal") val maal: Int = 0,
    @SerializedName("bonusPoint") val bonusPoint: Int = 0,
    @SerializedName("duply") val duply: Boolean = false,
    @SerializedName("winner") val winner: Boolean = false,
    @SerializedName("score") val score: Int = 0,
    @SerializedName("moneyWon") val moneyWon: Double = 0.0,
    @SerializedName("deal") val deal: Boolean = false,
    @SerializedName("position") val position: Int = 0
)
